tests/conftest.py

```diff
@@ -1703,10 +1703,11 @@ def titanic_sqlite_db(sa):
 
         titanic_db_path = file_relative_path(__file__, "./test_sets/titanic.db")
         engine = create_engine(f"sqlite:///{titanic_db_path}")
-        assert engine.execute(sa.text("select count(*) from titanic")).fetchall()[
-            0
-        ] == (1313,)
-        return engine
+        with engine.begin() as connection:
+            assert connection.execute(
+                sa.text("select count(*) from titanic")
+            ).fetchall()[0] == (1313,)
+            return connection
```

Comment:
Just looking closer at this - here we are beginning a transaction with engine.begin(). And the original method returns an engine. Should this instead be:
```suggestion
        with engine.connect() as connection:
            assert connection.execute(
                sa.text("select count(*) from titanic")
            ).fetchall()[0] == (1313,)
        return engine
```

Reply:
you're absolutely right :) thanks for the catch here.

Reply:
Sorry to re-hash this one, but can we return the engine instead of the connection (as it was previously?)